test/e2e/helpers/setupConnection.ts

```diff
@@ -1,32 +1,58 @@
+import { expect } from '@playwright/test'
 import 'reflect-metadata'
-import { fetchPost } from '../../helpers/routeHelper.js'
+
+import { fetchGet, fetchPost } from '../../helpers/routeHelper.js'
 import { checkEmails, extractInvite, extractPin, getHostPort } from './smtpEmails.js'
 
-export async function withConnection(smtp4devUrl: string, aliceUrl: string, bobUrl: string) {
-  // Send an invite from Alice to Bob
-  await fetchPost(`${aliceUrl}/connection/new/create-invitation`, {
+const uuidRegex = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89ab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}/g
+
+export async function withConnection(issuerUrl, holderUrl) {
+  const smtp4devUrl = process.env.VERITABLE_SMTP_ADDRESS || 'http://localhost:5001'
+
+  await fetchPost(`${issuerUrl}/connection/new/create-invitation`, {
     companyNumber: '04659351',
     email: 'alice@testmail.com',
     action: 'submit',
   })
 
-  // Get pin and invite
+  // Get pin and invite for the new holder
   const { host, port } = getHostPort(smtp4devUrl)
   if (host === null || port === null) {
     throw new Error(`Unspecified smtp4dev host or port ${smtp4devUrl}`)
   }
   const { inviteEmail, adminEmail } = await checkEmails(host, port)
-  const extractedPin = await extractPin(adminEmail.id, smtp4devUrl)
-  if (!extractedPin) throw new Error('PIN for Bob was not found.')
-  if (extractedPin.length !== 6) {
-    throw new Error('Pin does not have the expected length.')
-  }
-  const invite = await extractInvite(inviteEmail.id, smtp4devUrl)
-  if (!invite) throw new Error('Invitation for Bob was not found.')
+  const issuerPin = await extractPin(adminEmail.id, smtp4devUrl)
+  if (!issuerPin) throw new Error(`PIN for ${holderUrl} was not found.`)
+
+  const inviteBase64 = await extractInvite(inviteEmail.id, smtp4devUrl)
+  if (!inviteBase64) throw new Error('Invitation for Bob was not found.')
 
-  // Use invite and from Bob's side
-  await fetchPost(`${bobUrl}/connection/new/receive-invitation`, {
-    invite: invite,
+  // Verify and validate a new connection - holder
+  const receive = await fetchPost(`${holderUrl}/connection/new/receive-invitation`, {
+    invite: inviteBase64,
     action: 'createConnection',
   })
+  const [holderConnectionId] = (await receive.text()).match(uuidRegex) || []
+  await fetchPost(`${holderUrl}/connection/${holderConnectionId}/pin-submission`, {
+    action: 'submitPinCode',
+    pin: issuerPin,
```

Comment:
mabe the naming of issuer and holder is a bit confusing here? I understand as holder you want to submit pin and invite  FROM the issuer ...but this almost suggests to me this pin is for issuer to use ... @mattdean-digicatapult ?

Reply:
I think the terms are confudsing and I've made an alternative suggestion

Reply:
This has been updated to be `receiver` and `invitator`